boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -37,74 +51,119 @@ void App_SevenSegDisplays_Destroy(struct SevenSegDisplays *const seven_seg_displ
     free(seven_seg_displays);
 }
 
-ExitCode App_SevenSegDisplays_SetHexDigits(
-    const struct SevenSegDisplays *const seven_seg_displays,
-    const uint8_t                        hex_digits[],
-    size_t                               num_hex_digits)
+ExitCode
+    App_SevenSegDisplays_SetHexDigits(const struct SevenSegDisplays *seven_seg_displays, const uint8_t hex_digits[])
 {
-    if (num_hex_digits > NUM_SEVEN_SEG_DISPLAYS || num_hex_digits == 0)
-    {
-        return EXIT_CODE_INVALID_ARGS;
-    }
+    size_t len_hex_digits = 9; // TODO: This is bad, should get from somewhere else
 
-    // If any of the input digits is invalid, we don't write anything to the
-    // 7-segment displays at all.
-    for (size_t i = 0; i < num_hex_digits; i++)
+    for (size_t i = 0; i < len_hex_digits; i++)
     {
         if (hex_digits[i] >= NUM_HEX_DIGITS)
         {
             return EXIT_CODE_INVALID_ARGS;
         }
     }
 
-    for (size_t i = 0; i < NUM_SEVEN_SEG_DISPLAYS; i++)
+    return EXIT_CODE_OK;
+}
+
+void App_Set_Digits(const struct SevenSegDisplays *seven_seg_displays, uint8_t index)
+{
+    for (int i = index; i < NUM_IN_GROUP + index; i++)
     {
         struct SevenSegHexDigit hex_digit;
+        hex_digit.enabled = true;
+        hex_digit.value   = digits[i];
 
-        if (i < num_hex_digits)
-        {
-            hex_digit.enabled = true;
-            hex_digit.value   = hex_digits[i];
-        }
-        else
+        App_SevenSegDisplay_SetHexDigit(seven_seg_displays->displays[i], hex_digit, i);
+    }
+
+    seven_seg_displays->display_value_callback();
+}
+
+ExitCode App_SevenSegDisplays_SetGroupL(const struct SevenSegDisplays *const seven_seg_displays, uint32_t value)
+{
+    if (value > pow(10, NUM_IN_GROUP) - 1)
+    {
+        return EXIT_CODE_INVALID_ARGS;
+    }
+    digits[0] = 0;
+    digits[1] = 0;
+    digits[2] = 0;
```

Comment:
Can replace these indices with the digit enum values. In general I think magic numbers like this are usually a bad idea